### PR TITLE
refactor-completed-jobs-endpoint

### PR DESCRIPTION
**Pull request by oanaBiancaSchiopu**
Remove timeWindow parameter
Add startTime, endTime and numberOfIntervals parameters

### CODE REVIEW

File: rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java [hunk 1]
```diff
@@ -2190,8 +2191,9 @@ List<WorkflowDurationData> getTopPendingTimeWorkflows(@HeaderParam("sessionid")
     CompletedJobsCountData getCompletedJobs(@HeaderParam("sessionid") String sessionId,
             @QueryParam("myjobs") @DefaultValue("false") boolean myJobs,
             @QueryParam("workflowName") @DefaultValue("null") String workflowName,
-            @QueryParam("timeWindow") String timeWindow, @QueryParam("zoneId") @DefaultValue("null") String zoneId)
-            throws RestException;
+            @QueryParam("startdate") @DefaultValue("0") long startDate,
+            @QueryParam("enddate") @DefaultValue("0") long endDate,
+            @QueryParam("numberOfIntervals") int numberOfIntervals) throws RestException;
```
Comment (fviale) [inline rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java:2196]: I would prefer if endDate default value -1 with a meaning of now() (this should be implemented in the code).
numberOfIntervals default value should be 1

File: rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java [hunk 2]
```diff
@@ -2696,12 +2696,13 @@ public List<WorkflowDurationData> getTopPendingTimeWorkflows(@HeaderParam("sessi
     @Override
     public CompletedJobsCountData getCompletedJobs(@HeaderParam("sessionid") String sessionId,
             @QueryParam("myjobs") @DefaultValue("false") boolean myJobs,
-            @QueryParam("workflowName") String workflowName, @QueryParam("timeWindow") String timeWindow,
-            @QueryParam("zoneId") String zoneId) throws RestException {
+            @QueryParam("workflowName") String workflowName, @QueryParam("startdate") @DefaultValue("0") long startDate,
```
Comment (fviale) [inline rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java:2699]: same remark concerning endDate default value

File: rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java [hunk 3]
```diff
@@ -2696,12 +2696,13 @@ public List<WorkflowDurationData> getTopPendingTimeWorkflows(@HeaderParam("sessi
...
+            @QueryParam("enddate") @DefaultValue("0") long endDate,
+            @QueryParam("numberOfIntervals") int numberOfIntervals) throws RestException {
```
Comment (fviale) [inline rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java:2701]: same remarks as above

**Review by tao-xinxiu — APPROVED** [review]

File: rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java [hunk 4]
```diff
@@ -2179,8 +2179,9 @@ List<WorkflowDurationData> getTopPendingTimeWorkflows(@HeaderParam("sessionid")
      * @param sessionId id of a session
      * @param myJobs fetch only the jobs owned by the user making the request
      * @param workflowName the workflow name of the filtered jobs
-     * @param timeWindow the statistics time window: daily, weekly or monthly
-     * @param zoneId the requested zone id, e.g. Europe/Paris. If it remains empty, the zone id will be the same as the server zone
+     * @param startDate start date of the filtered jobs
+     * @param endDate end date of the filtered jobs
```
Comment (fviale) [inline rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java:2183]: you should indicate that the default value of -1 means until now

**Pull request merged**